src/BetterAppListTrait.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\apigee_edge_ui;
+
+/**
+ * Copyright (C) 2020 PRONOVIX GROUP BVBA.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
+ * USA.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\StringTranslation\TranslatableMarkup;
+use Drupal\Core\Template\Attribute;
+use Drupal\apigee_edge\Entity\AppInterface;
+
+/**
+ * List builder additions for apps.
+ */
+trait BetterAppListTrait {
+
+  /**
+   * Alters the app info in the original list render array.
+   *
+   * @param array $build
+   *   The original render array.
+   * @param string $rel
+   *   The app name's link relationship type, defaults to 'canonical'.
+   */
+  private function buildAppListContent(array &$build, string $rel = 'canonical'): void {
+    // Use custom template instead of table.
+    unset($build['table']['#type']);
+    if (isset($this->entityTypeId)) {
+      $build['table']['#type'] = $this->entityTypeId;
+    }
+    $build['table']['#theme'] = 'apigee_edge_ui_list';
+
+    // Build the app rows from scratch.
+    $build['table']['#items'] = [];
+    foreach ($this->load() as $entity) {
+      /** @var \Drupal\apigee_edge\Entity\AppInterface $entity */
+      $app_row = [
+        '#attributes' => new Attribute([
+          'id' => $this->getCssIdForInfoRow($entity),
+          'class' => 'row--info',
+        ]),
+        'name' => [
+          '#type' => 'link',
+          '#title' => $entity->label(),
+          '#url' => $entity->toUrl($rel),
+        ],
+        'status' => $this->renderAppStatus($entity),
+        'operations' => $this->buildOperations($entity),
+        '#warning_attributes' => new Attribute([
+          'id' => $this->getCssIdForWarningRow($entity),
+          'class' => 'row--warning',
+        ]),
+      ];
+
+      if (isset($build['table']['#header']['team'])) {
+        /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface[] $teams */
+        $teams = $this->entityTypeManager->getStorage('team')->loadMultiple();
```

Comment:
Why do you load all teams instead of the one that you need?  ($this->entityTypeManager is also not defined/required by the trait so it can be null anytime)